feat: app center migration cli tool (#563)
appcircleio/appcircle-docusaurus: docs/appcircle-api/appcenter-migration-tool.md

---
title: App Center Migration Tool
description: Migrate App Center organizations, collaborators,  apps to testing distribution profiles and test groups
tags: [app center, migration, cli]
sidebar_position: 3
---

import Screenshot from '@site/src/components/Screenshot';

The appcenter-migration-tool is designed to assist App Center projects in seamlessly migrating their organizations, collaborators, app profiles, and test groups to Appcircle.

By using the **appcenter-migration-tool**, you can ensure a smooth and efficient migration process, minimizing downtime and preserving the integrity of your data.

**Key Features**:

- **Organization Migration:** Effortlessly transfer your entire organization structure to the new platform.

- **Collaborator Migration:** Move all your collaborators with their roles and permissions intact.

- **App Profile Migration:** Migrate app profiles, ensuring all settings and configurations are preserved.

- **Test Group Migration:** Transition your test groups with all their associated configurations and data.

> By using the appcenter-migration-tool, you can ensure a smooth and efficient migration process, minimizing downtime and preserving the integrity of your data.

Reviewer: we can delete this part as it's duplicated.

| ---------------------------------------------------------------- | ----------------- | ---------------- | ----------------------------------------------------------------------------------------------------------------------- |
| List All App Center Apps                                         | list              | -                | List All App Center Apps.                                                                                               |
| Migrate App Center App to Appcircle Testing Distribution Profile | list-organization | organizationName | List App Center Apps Based on App Center Organization                                                                   |
| Migrate App Center App to Appcircle Testing Distribution Profile | migrate-profile   | profileName      | Migrate App Center App to Appcircle [Testing Distribution Profile](/distribute/create-or-select-a-distribution-profile) |

To run a command directly instead of entering an interactive session, simply execute the command as shown below:

```bash
appcenter-migration-tool apps list
```

:::caution

> migrate-profile command only creates a **Testing Distribution Profile** with the app name at Appcircle. Currently Build details does not included in migration

Reviewer: ```suggestion
migrate-profile command only creates a **Testing Distribution Profile** with the app name at Appcircle. Currently, Build details are not included in migration.
```

| Migrate App Center App Distribution Group to Appcircle                | migrate-app          | organizationName appName distributionGroupNameForApp distGroupUsersForApp | Migrate App Center App Distribution Group to Appcircle                     |

To run a command directly instead of entering an interactive session, simply execute the command as shown below:

```bash
appcenter-migration-tool distribution-groups list-organization --organizationName=YOUR_ORGANIZATION_NAME
```

## Migrate Sample Organization

In this section, we will provide a comprehensive overview of the migration process from App Center to Appcircle. App Center and Appcircle manage organizations and distribution groups slightly differently. The following diagram illustrates a sample organization structure in both App Center and Appcircle.

**Organization Hierarchy for Migration:**

App Center organization hierarchy slated for migration is structured as follows:

<Screenshot url='https://cdn.appcircle.io/docs/assets/SP-194_appcenter_hierarchy.png' />

Migrated Appcircle organization is structured as follows:

<Screenshot url='https://cdn.appcircle.io/docs/assets/SP-194_appcircle_hierarchy.png' />

> Demonstrates how migration might look like step-by-step:
> 
> 1. Migrating the organizations from App Center to Appcircle. Appcircle manages organizations using a main-sub organization structure. This means that every organization you migrate will be part of a main organization.
> 2. Migrating apps from App Center to Appcircle involves creating a Testing Distribution Profile in Appcircle. This process currently includes only the creation of the profile; **migration of build configuration details is not supported yet**.
> 3. Migrating Distribution Groups from App Center to Appcircle. App Center allows users to manage distribution groups in two ways: at the organization level and at the app level. In Appcircle, distribution groups are managed exclusively at the organization level.

Reviewer: ```suggestion

**Step-by-Step Migration Process:**

1. **Migrating Organizations:** 
    - Migrate the organizations from App Center to Appcircle. 
    - Appcircle uses a main-sub organization structure, meaning every organization you migrate will become part of a main organization.

2. **Migrating Apps:**
    - Create a Testing Distribution Profile in Appcircle for each app from App Center.
    - This process currently includes only the creation of the profile; migration of build configuration details is not supported yet.

3. **Migrating Distribution Groups:**
    - Migrate Distribution Groups from App Center to Appcircle.
    - App Center allows users to manage distribution groups at both the organization level and the app level.
    - In Appcircle, distribution groups are managed exclusively at the organization level.

```
